pkg/m3db/reconcile_status.go

```diff
@@ -0,0 +1,27 @@
+package m3db
+
+import (
+	"reflect"
+
+	api "github.com/m3db/m3db-operator/pkg/apis/m3db/v1alpha1"
+
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+)
+
+func updateM3DBStatus(s *api.M3DBService, status *api.M3DBServiceStatus) error {
+	// don't update the status if there aren't any changes.
+	if reflect.DeepEqual(s.Status, *status) {
+		return nil
+	}
+	s.Status = *status
+	return sdk.Update(s)
+}
+
+func getM3DBStatus(s *api.M3DBService) (*api.M3DBServiceStatus, error) {
```

Comment:
these two functions can return errors, but I don't see error-checking implemented, is that TBD?

Reply:
Currently there isn't any error-able code at the moment.